src/base/utils/net.cpp

```diff
@@ -89,5 +90,25 @@ namespace Utils
         {
             return subnet.first.toString() + '/' + QString::number(subnet.second);
         }
+
+        boost::optional<QList<QSslCertificate>> tryLoadSSLCertificate(const QByteArray &data)
+        {
+            const QList<QSslCertificate> certs {QSslCertificate::fromData(data)};
+            if (certs.isEmpty()
+                || std::any_of(certs.cbegin(), certs.cend(), [](const QSslCertificate &c) { return c.isNull(); }))
+                return {};
+            return certs;
+        }
+
+        boost::optional<QSslKey> tryLoadSSLKey(const QByteArray &data)
```

Comment:
The same comment as above. Either throw an error or return "empty" QSslKey.

Reply:
Also I strongly dislike function names like "tryDoSomething()". "try" is totally meaningless prefix. It is clear that some functions may not be successful. It follows from their meaning (loading, parsing etc. routines), it is not necessary to specify it in addition.

Reply:
will remove the prefix.